src/main/java/appeng/api/networking/crafting/ICraftingGrid.java

```diff
@@ -56,7 +56,7 @@ ImmutableCollection<ICraftingPatternDetails> getCraftingFor(IAEItemStack whatToC
      *         on this, your be waiting forever.
      */
     Future<ICraftingJob> beginCraftingJob(World world, IGrid grid, BaseActionSource actionSrc, IAEItemStack craftWhat,
-            ICraftingCallback callback);
+            boolean usePreCrafting, ICraftingCallback callback);
```

Comment:
This is a public API class, add a default method with the old signature that forwards the call without pre-crafting. to avoid breaking existing users and the extra param in javadocs. 
Also instead of a boolean, make a public enum CraftingMode {STANDARD, IGNORE_MISSING} for example (I think ignore missing is a better name than precrafting)